### PR TITLE
Update IJSValueReader and IJSValueWriter docs

### PR DESCRIPTION
**Pull request by vmoroz**
The updated files are auto-generated from the document comment changes to `IJSValueReader.idl' and `IJSValueWriter.idl' files.
These files are generated from the doc comment changes in PR https://github.com/microsoft/react-native-windows/pull/6891.

### CODE REVIEW

**Review by NikoAri — APPROVED** [review]
:shipit:

File: docs/IJSValueWriter-api-windows.md [hunk 1]
```diff
@@ -7,78 +7,78 @@ Kind: `interface`
 
 
 
-`IJSValueWriter` is used to write JavaScript values from custom native modules.  It acts as a stream and supports all the types in [`JSValueType`](JSValueType).
+JSON-like stream writer.<br/>It is used to write data that is sent between native modules and the Microsoft.ReactNative.dll.<br/><br/>The JSON-like streams are data structures that satisfy the [JSON specification](https://tools.ietf.org/html/rfc8259). The data structure may have objects with name-value pairs and arrays of items. Property values or array items can be of type `Null`, `Object`, `Array`, `String`, `Boolean`, or `Number`. The `IJSValueWriter` treats the `Number` type as `Int64` or `Double`. See [`JSValueType`](JSValueType).<br/><br/>See the [`IJSValueReader`](IJSValueReader) for the corresponding reader interface.<br/><br/>The [`IJSValueReader`](IJSValueReader) and [`IJSValueWriter`](IJSValueWriter) should be rarely used directly. They are usually used to create serializer and deserializer functions for a type. In the rest of our code we must use these functions to serialize/deserialize our values. We already have a set of serializer and deserializer functions for some standard types.
```
Comment (asklar) [inline docs/IJSValueWriter-api-windows.md:10]: > In the rest of our code

Whose code does this refer to? RNW? app code?

File: docs/JSValueType-api-windows.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ title: JSValueType
 
 Kind: `enum`
 
+Type of value read by IJSValueReader.
```
Comment (asklar) [inline docs/JSValueType-api-windows.md:8]: consider referencing `@IJSValueReader` instead

File: docs/IJSValueWriter-api-windows.md (hunk 1 above)
Comment (vmoroz) [reply docs/IJSValueWriter-api-windows.md]: Updated

File: docs/JSValueType-api-windows.md (hunk 2 above)
Comment (vmoroz) [reply docs/JSValueType-api-windows.md]: Updated.

File: docs/IJSValueReader-api-windows.md [hunk 3]
```diff
@@ -7,56 +7,56 @@ Kind: `interface`
...
-Gets the current boolean value.
+Gets the current `Boolean` value.
 
 
 
 ### GetDouble
 double **`GetDouble`**()
 
-Gets the current number value as a double.
+Gets the current `Number` value as a `Double`.
```
Comment (stmoy) [inline docs/IJSValueReader-api-windows.md:31]: Quick clarifying question - is there a difference between `double` and `Double`? I _think_ there is (at least in C#) where one is a primitive type and the other is an object, but I don't know if it matters here or not.

Comment (vmoroz) [reply docs/IJSValueReader-api-windows.md:31]: I think in this context there is no difference. I have changed to use the same terms as in the API.

**Pull request merged**